scylla/src/utils/futures.rs

```diff
@@ -0,0 +1,4 @@
+use std::future::Future;
+use std::pin::Pin;
+
+pub(crate) type BoxedFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
```

Comment:
It's not a pub item but it appears in public traits. How does it interact with the documentation? I'd rather users didn't encounter an opaque `BoxFuture` alias. Users should know what is the definition, that it requires the future to be `Send` etc. - I think this information should be available if you make `BoxedFuture` public.

Reply:
Yes, you are right. From what I've tested the users trying to implement the trait encounter an opaque type (`BoxedFuture`) and cannot import it (since the definition is `pub(crate)`).